backup/metadata.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package backup
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/version"
+
+	"github.com/juju/juju-restore/core"
+)
+
+func readMetadataJSON(path string) (core.BackupMetadata, error) {
+	source, err := os.Open(path)
+	if err != nil {
+		return core.BackupMetadata{}, errors.Trace(err)
+	}
+	defer source.Close()
+	data, err := ioutil.ReadAll(source)
+
+	// Try the current version and check the FormatVersion first.
+	var target flatMetadata
+	err = json.Unmarshal(data, &target)
+	if err != nil {
+		return core.BackupMetadata{}, errors.Annotate(err, "unmarshalling v1 metadata")
+	}
+
+	if target.FormatVersion == 1 {
+		return flatToBackupMetadata(target), nil
+	}
+
+	// No FormatVersion set - it must be a version 0 structure
```

Comment:
This will fail when there is version 2?.. Maybe worth having a switch where case 0 and 1 is cater for and default says "Unsupported"?

Reply:
Oh right - yes, this should definitely fail for version < 1! Adding that now.